src/main.rs

```diff
@@ -9,58 +10,144 @@ struct Player {
     five_in_a_row: bool
 }
 
-fn print_board(board: Vec<Vec<char>>){
-    println!("Contents of the board:");
+#[derive(Clone)]
+enum BoardVal {
+    Empty,
+    Player1,
+    Player2,
+}
+
+fn print_board(board: Vec<Vec<BoardVal>>) {
+    // print column identifiers
+    println!();
+    print!("{:3}", 0);
+    for col in 1..board.len() {
+        print!("{:4}", col);
+    }   
+    println!();
+
+    // print board rows
     for row in 0..board.len() {
-        for col in 0..board[row].len(){
-            print!("{}", board[row][col]);
+        // print row identifier
+        print!("{} ", (65 + row) as u8 as char);
+
+        // print row contents
+        for col in 0..board[row].len() {
+            print!("{}", 
+                match board[row][col] {
+                    BoardVal::Empty => {
+                        if row == 0 {
+                            if col == 0 { "┌" }   
+                            else if col == board.len()-1 { "┐" } 
+                            else { "┬" }
+                        }
+                        else if row == board.len()-1 {
+                            if col == 0 { "└" }   
+                            else if col == board.len()-1 { "┘" } 
+                            else { "┴" }
+                        } 
+                        else if col == 0 { "├" }
+                        else if col == board.len()-1 { "┤" }
+                        else { "┼" }
+                    },
+                    BoardVal::Player1 => "◎",
+                    BoardVal::Player2 => "●",
+                }
+            );
+            // don't print filler if on last column
+            if col == board.len()-1 { continue; }
+            print!("───");
         }
-        println!("");
+        println!();
+
+        // if on last row, don't print blank line underneath
+        if row == board.len()-1 { break; }
+
+        // print blank lines
+        print!("  ");
+        for _col in 0..board[row].len() {
```

Comment:
I was getting a warning bc `col` was unused, so I replaced with `_col`. Was it intentional to not do this? (sorry I didn't mean to actually change that in this commit, but I had added it locally to remove the error I was getting).